src/device.coffee

```diff
@@ -10,36 +10,36 @@ execAsync = Promise.promisify(require('child_process').exec)
 fs = Promise.promisifyAll(require('fs'))
 bootstrap = require './bootstrap'
 
-exports.getID = do ->
-	deviceIdPromise = null
-	return ->
-		# We initialise the rejected promise just before we catch in order to avoid a useless first unhandled error warning.
-		deviceIdPromise ?= Promise.rejected()
-		# Only fetch the device id once (when successful, otherwise retry for each request)
-		deviceIdPromise = deviceIdPromise.catch ->
-			# Wait for bootstrapping to be done before querying the Resin API
-			# This will also block users of getID (like applyState below until this is resolved)
-			bootstrap.done
-			.then ->
-				Promise.all([
-					knex('config').select('value').where(key: 'apiKey')
-					knex('config').select('value').where(key: 'uuid')
-				])
-			.spread ([{ value: apiKey }], [{ value: uuid }]) ->
-				resinApi.get(
-					resource: 'device'
-					options:
-						select: 'id'
-						filter:
-							uuid: uuid
-					customOptions:
-						apikey: apiKey
-				)
-				.timeout(config.apiTimeout)
-			.then (devices) ->
-				if devices.length is 0
-					throw new Error('Could not find this device?!')
-				return devices[0].id
+memoizePromise = (func) ->
```

Comment:
This can be replaced by memoizee https://www.npmjs.com/package/memoizee#promise-returning-functions
```coffee
memoizePromise = _.partial(memoizee, _, promise: true)
```
which avoids having custom code for this stuff and can also be used for more advanced memoisation if we need it - but mainly it's just to reduce the amount of code we need to maintain here